Monika After Story/game/script-apologies.rpy

```diff
@@ -0,0 +1,250 @@
+    #No need to store excess memory
+    $ del player_apology_reasons
+
+    #Now we run through our apology db and find what's unlocked
+    python:
+        apologylist = [
+            (mas_getEV(ev).prompt,ev,False,False)
+            for ev in store.mas_apology.apology_db
+            if mas_getEV(ev).unlocked
+        ]
+
+        #The back button
+        return_prompt_back = ("Nevermind", False, False, False, 20)
+
+    #Display our scrollable
+    show monika at t21
+    call screen mas_gen_scrollable_menu(apologylist,(evhand.UNSE_X, evhand.UNSE_Y, evhand.UNSE_W, 500), evhand.UNSE_XALIGN, return_prompt_back)
+    show monika at t11
+
+    #Make sure we don't lose this value
+    $ apology =_return
+
+    #Handle backing out
+    if not apology:
+        if mas_apology_reason is not None or persistent._mas_apology_time_db != {}:
+            if mas_isMoniAff(higher=True):
+                m 1ekd "[player], if you're feeling guilty about what happened..."
+                m 1eka "You don't have to be afraid of apologizing, we all make mistakes after all."
+                m 3eka "We just have to accept what happened, learn from our mistakes, and move on, together. Okay?"
+            elif mas_isMoniNormal(higher=True):
+                m 1eka "[player]..."
+                m "If you want to apologize, go ahead. It'd mean a lot to me if you did."
+            elif mas_isMoniUpset():
+                m 2rkc "Oh..."
+                m "I was kind of--"
+                $ _history_list.pop()
+                m 2dkc "Nevermind."
+            elif mas_isMoniDis():
+                m 6rkc "...?"
+            else:
+                m 6ckc "..."
+        else:
+            if mas_isMoniAff(higher=True):
+                m 1hua "Ehehe, you don't have anything to be sorry about silly~"
+            elif mas_isMoniNormal(higher=True):
+                m 1eka "I'm not sure what you wanted to apologize for, but I'm sure that it was nothing."
+
+        return
+
+
+    #Call our apology label
+    #NOTE: mas_setApologyReason() ensures that this label exists
+    $ renpy.call(apology)
+
+    #Increment the shown count
+    $ mas_getEV(apology).shown_count += 1
+
+    #Lock the apology label if it's not the generic
+    if not apology == "mas_apology_generic":
+        $ store.mas_hideEVL(apology,'APL',lock=True)
+
+    #Pop that apology from the time db
+    if apology in persistent._mas_apology_time_db: #sanity check
+        $ persistent._mas_apology_time_db.pop(apology)
+    return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.apology_database,
+            prompt="...for something else.",
+            eventlabel="mas_apology_generic",
+            unlocked=True,
+        ),
+        code="APL"
+    )
+
+label mas_apology_generic:
+    #dict of all generic apologies
+    #Note, if a custom apology is needed, add it here and reference the apology reason by the integer associated.
+    $ mas_apology_reason_db = {
```

Comment:
Is this the same as the one in playerapologizes?

If so, move this to an init store. You can define it in a function that can get the apology reason given the id  if you're aiming to save on memory.

Reply:
This one is similar to the one in playerapologizes, this one stores what Monika responds, while the one in playerapolgizes stores the prompts for the generic, so it's in the perspective of the player's voice.